astropy/visualization/wcsaxes/wcsapi.py

```diff
@@ -100,6 +100,22 @@ def transform_coord_meta_from_wcs(wcs, frame_class, aslice=None):
     coord_meta['default_ticklabel_position'] = [''] * wcs.world_n_dim
     coord_meta['default_ticks_position'] = [''] * wcs.world_n_dim
 
+    invert_xy = False
+    if slices is not None:
+        wcs_slice = list(slices)
+        wcs_slice[wcs_slice.index("x")] = slice(None)
+        wcs_slice[wcs_slice.index("y")] = slice(None)
+        wcs = SlicedLowLevelWCS(wcs, wcs_slice[::-1])
+        invert_xy = slices.index('x') > slices.index('y')
+        world_keep = wcs._world_keep
```

Comment:
@Cadair - this is a little ugly but I can't think of another way to do it. This is probably ok pragmatically since we are within the same overall package, but I wonder if we might want to somehow expose this information in a public API (maybe in a separate PR).

Reply:
yeah makes sense on both fronts.